target/bin/rspamd-dkim

```diff
@@ -253,7 +246,15 @@ domain {
 }
 
 EOF
-    chown _rspamd:_rspamd "${DEFAULT_CONFIG_FILE}"
+
+    # We copy here immediately in order to not rely on the changedetector - this way, users
+    # can immediately use the new keys. The file should not already exist in ${RSPAMD_OVERRIDE_D}
+    # since it would have been copied already.
+    cp "${DEFAULT_CONFIG_FILE}" "${RSPAMD_OVERRIDE_D}/dkim_signing.conf"
+    chown _rspamd:_rspamd "${DEFAULT_CONFIG_FILE}" "${RSPAMD_OVERRIDE_D}/dkim_signing.conf"
+
+    _log 'debug' 'Restarting Rspamd as initial DKIM configuration was suppplied'
+    supervisorctl restart rspamd
```

Comment:
> This will still potentially trigger a change detection right? As it compares to it's before/after checksums in the config volume?

`STAGING_FILES` are crated by scanning `RSPAMD_DMS_D` (i.e. `/tmp/docker-mailserver/rspamd`), so AFAIK it should not trigger another event (since the `cp` copies to a location not checked and `chown` does not change SHA sum IIRC.


> It seems that DKIM keys aren't copied over to an internal location like the certs are? I'm not too concerned about that I guess 🤔

Yes. This could be changed in the future if we encounter problems, but it seems to be absolutely fine for now.

> FWIW, the rspamd docs mention [rspamadm template](https://rspamd.com/doc/rspamadm.html) for generating configs from jinja templates if you wanted to take advantage of that 🤷‍♂️

Have not looked intot that yes.

Reply:
**About the Rspamd DKIM proposals**: wow, that was very interesting! Funnily enough, I think we should first try the `.include ...` macros before dancing around with `yq` (even though I found the `yq` stuff cool). Thank you for sharing your ideas here. I will make sure to remember this location; the reason I didn't touch DKIM for Rspamd here any more than the `--force` option is that the PR is already pretty big. The `--force` option was due to a recent issue I stumbled upon. I know you'd like to merge the OpenDKIM and Rspamd DKIM scripts, and I can tackle this in the future; for now, I'd like this PR to stay as-is to not inflate it any more. For a future PR that tackles DKIM again, we back-reference to your comments here!

Reply:
> I know you'd like to merge the OpenDKIM and Rspamd DKIM scripts, and I can tackle this in the future; for now, I'd like this PR to stay as-is to not inflate it any more.
> For a future PR that tackles DKIM again, we back-reference to your comments here!

Oh yeah I didn't mean to update this PR to apply the review comments here, definitely a separate PR 👍 

I agree the include macro would be wiser. I'd have suggested that before typing up all the `yq` stuff if I had read the rest of the UCL readme first.... 😓